msticpy/init/pivot_core/pivot_pd_accessor.py

```diff
@@ -442,6 +442,37 @@ def parse_json(self, cols: Union[str, Iterable[str]]) -> pd.DataFrame:
             data = data.drop([col], axis=1).rename(columns={col_parsed: col})
         return data
 
+    def json_to_pandas(self, col: str) -> pd.DataFrame:
```

Comment:
Maybe dict_to_dataframe would be a better name.
You could also test the column type - it could be a Python dict (which your code handles) or it could be a json string - if it's a string you could try json.decode(record)
```
for row_value in self._df[col]:
  if isinstance(row_value, dict):
    record = row_value
  elif isinstance(row_value, str):
    try:
       record = json.decode(row_value)
    except json.JSONDecodeError:
       continue
....
```